src/Lunr/Vortex/APNS/ApnsPHP/APNSDispatcher.php

```diff
@@ -221,11 +114,90 @@ public function push(object $payload, array &$endpoints): APNSResponse
         }
 
         // Return response
-        $response = new APNSResponse($this->logger, $endpoints, $invalid_endpoints, $errors, (string) $this->apns_message);
+        return new APNSResponse($this->logger, $endpoints, $invalid_endpoints, $errors, (string) $message);
+    }
+
+    /**
+     * Build a Message object for a given payload
+     *
+     * @param APNSPayload $payload The payload to build from
+     *
+     * @return Message The filled message
+     */
+    private function build_message_for_payload(APNSPayload $payload): Message
+    {
+        $payload = $payload->get_payload();
+        $message = new Message();
```

Comment:
This breaks dependency injection principles. The only way to replace it with another object is by replacing the entire message construction.